src/types/config.js

```diff
@@ -88,18 +111,65 @@
+ * Settings derived from `AnnotatorConfigFromHost["requestConfigFromFrame"]`.
+ * These settings allow `EmbedderConfigFromHost` to be requested from the
+ * designated frame, and allows subsequent communication between the sidebar
+ * and embedder frames.
  *
  * @typedef RPCSettings
  * @prop {Window} targetFrame
- * @prop {RequestConfigFromFrameOptions['origin']} origin
+ * @prop {EmbedderFrameConfig['origin']} origin
  */
 
 /**
- * The `settings` object used in the sidebar app that is a result of merging
- * (filtered and validated) configuration from the host page with configuration
- * from h / the browser extension.
+ * `SidebarSettings` are created by merging `ConfigWithSidebar` with appropriate
+ * `ConfigFromHost` ("host configuration").
+ *
+ * In all cases, the annotator ("host frame") provides `AnnotatorConfigFromHost`
+ * to the sidebar frame by encoding values into a URL fragment on the sidebar
+ * frame's `src` attribute.
+ *
+ * In most cases, `AnnotatorConfigFromHost` is used as `ConfigFromHost`:
+ *
+ * +--------------------------------------------+
+ * |        host frame (annotator)              |
+ * |                 +-----------------------+  |
+ * |                 |  sidebar frame        |  |
+ * |                 |                       |  |
+ * | <AnnotatorConfigFromHost> => iframe.src |  |
+ * |                 |                       |  |
+ * |                 |                       |  |
+ * |                 |                       |  |
+ * |                 | <ConfigFromSidebar>   |  |
+ * |                 +-----------------------+  |
+ * +--------------------------------------------+
+ *
+ * In some cases (e.g. LMS), host configuration should instead be provided by an
+ * ancestor ("embedder") frame. This is signaled by the presence of a valid
+ * `EmbedderFrameConfig` object on `AnnotatorConfigFromHost`.
+ *
+ * `EmbedderConfigFromHost` will then be requested from the designated embedder
+ * frame and used as `ConfigFromHost`:
+ *
+ * +------------------------------------------------------------------------+
+ * |                                         embedder frame                 |
+ * |  +------------------------------------------+                          |
+ * |  |        host frame (annotator)            |                          |
+ * |  |                 +---------------------+  |                          |
+ * |  |                 |  sidebar frame      |  |                          |
+ * |  |                 |                     |  |                          |
+ * |  |<AnnotatorConfigFromHost> => iframe.src|  |                          |
+ * |  |   EmbedderFrameConfig                 |  |                          |
+ * |  |                 |                     |  |                          |
+ * |  |                 |                     |  |                          |
+ * |  |                 |           <===JSONRPC===> <EmbedderConfigFromHost>|
```

Comment:
```suggestion
 * |  |                 |           <=== postMessage ===> <EmbedderConfigFromHost>|
```

JSON-RPC is the protocol, but the browser's `postMessage` APIs are the transport, which I think is probably more important to understand to get how this all works. We could for example, change JSON-RPC to some other protocol without changing the overall picture at all.

Reply:
> JSON-RPC is the protocol, but the browser's postMessage APIs are the transport,

Thank you. We (ahem, I) should definitely take more care with terminology.